src/coreclr/debug/inc/riscv64/primitives.h

```diff
@@ -0,0 +1,259 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+//*****************************************************************************
+// File: primitives.h
+//
+
+//
+// Platform-specific debugger primitives
+//
+//*****************************************************************************
+
+#ifndef PRIMITIVES_H_
+#define PRIMITIVES_H_
+
+typedef const BYTE                  CORDB_ADDRESS_TYPE;
+typedef DPTR(CORDB_ADDRESS_TYPE)    PTR_CORDB_ADDRESS_TYPE;
+
+#define MAX_INSTRUCTION_LENGTH 4
+
+// Given a return address retrieved during stackwalk,
+// this is the offset by which it should be decremented to land at the call instruction.
+#define STACKWALK_CONTROLPC_ADJUST_OFFSET 4
+
+#define PRD_TYPE                               LONG
+#define CORDbg_BREAK_INSTRUCTION_SIZE 4
+#define CORDbg_BREAK_INSTRUCTION (LONG)0x00100073
+
+inline CORDB_ADDRESS GetPatchEndAddr(CORDB_ADDRESS patchAddr)
+{
+    LIMITED_METHOD_DAC_CONTRACT;
+    return patchAddr + CORDbg_BREAK_INSTRUCTION_SIZE;
+}
+
+#define InitializePRDToBreakInst(_pPRD)       *(_pPRD) = CORDbg_BREAK_INSTRUCTION
+#define PRDIsBreakInst(_pPRD)                 (*(_pPRD) == CORDbg_BREAK_INSTRUCTION)
+
+
+#define CORDbgGetInstructionEx(_buffer, _requestedAddr, _patchAddr, _dummy1, _dummy2)                          \
+    CORDbgGetInstructionExImpl((CORDB_ADDRESS_TYPE *)((_buffer) + (_patchAddr) - (_requestedAddr)));
+
+#define CORDbgSetInstructionEx(_buffer, _requestedAddr, _patchAddr, _opcode, _dummy2)                          \
+    CORDbgSetInstructionExImpl((CORDB_ADDRESS_TYPE *)((_buffer) + (_patchAddr) - (_requestedAddr)), (_opcode));
+
+#define CORDbgInsertBreakpointEx(_buffer, _requestedAddr, _patchAddr, _dummy1, _dummy2)                        \
+    CORDbgInsertBreakpointExImpl((CORDB_ADDRESS_TYPE *)((_buffer) + (_patchAddr) - (_requestedAddr)));
+
+
+constexpr CorDebugRegister g_JITToCorDbgReg[] =
+{
+    REGISTER_RISCV64_RA,
```

Comment:
I added REGISTER_RISCV64_PC which is not directly accessed instead of REGISTER_RISCV64_X0.
If I try to add REGISTER_RISCV64_X0, but it asserts because it cannot exceed 64 registers (INT + FLOAT registers).
So I add TODO comment now. Thank you!

Reply:
What I meant was that `g_JITToCorDbgReg` seems to be accessed by reg index, i.e. `g_JITToCorDbgReg[0]` means `REGISTER_RISCV64_X0`, `g_JITToCorDbgReg[1]` means `REGISTER_RISCV64_RA` and so on (and same on other platforms). But current implementation will need to use `g_JITToCorDbgReg[reg-1]`.

Reply:
I see. As you mentioned, it can be a problem.
I added a X0 and put it at first location in g_JITToCorDbgReg. And added a TODO comment.
Actually, I copied loongarch codes and updated register names only. I think it needs to update loongarch too.